cranelift-codegen/meta/src/cdsl/encodings.rs

```diff
@@ -84,6 +84,24 @@ impl EncodingBuilder {
                     inst_predicate = Some(type_predicate.into());
                 }
 
+                // Add immediate value predicates
+                for (immediate_value, immediate_operand) in inst
+                    .immediate_values
+                    .iter()
+                    .zip(inst.inst.operands_in.iter().filter(|o| o.is_immediate()))
+                {
+                    let immediate_predicate = InstructionPredicate::new_is_field_equal(
+                        &inst.inst.format, // TODO need a handle to a FormatRegistry here
```

Comment:
One issue is that in order to use this type of predicate we would need a handle to a `FormatRegistry`; another issue is that I believe this `is_field_equal` predicate is actually doing a string comparison, which is a compile-time cost but a cost nonetheless.

Reply:
The EncodingBuilder are mostly used in the encodings.rs files, where there are also stores for all the encodings per mode; we could store the FormatRegistry by ref there, and pass it to this function.

is_field_equal ends up generating a `predicates::is_equal` call which handles any types that implements Eq, so we should be fine for those.